xtask/src/main.rs

```diff
@@ -16,12 +17,58 @@ fn print_help() {
         "Tasks:
 
 gen_enum_defaults generates Default trait impls for enums
+gen_constants generates constants used for Conditions
 "
     )
 }
 
 type DynError = Box<dyn std::error::Error>;
 
+fn gen_condition_constants() -> Result<(), DynError> {
+    let gateway_condition_types = env::var("GATEWAY_CONDITION_CONSTANTS")?;
```

Comment:
It occurs to me that it would be more ideal if we derived these values from the upstream CRDs. If nothing else, I think it would be good to leave behind an issue suggesting that we could try and emit datapoints into the CRDs in the future that this and other client libraries could then use for derivation.

Reply:
it'd be 100% more ideal to derive these values from the upstream CRDS, but i don't think the yamls have these constants, they exist solely in the Go APIs.
lets open an issue about emitting datapoints into the CRDs. ideally, we'd want the property to have an enum definition using the `+kubebuilder:validation:Enum` annptation, but since the `Condition` object is imported from upstream client-go, that won't be possible.

Reply:
Right that's what I'm saying let's open issue(s) about a follow up like that, and then please consider this comment resolved once we've done that and let's :ship: